File: ssr-simple/src/main/java/com/cincinnatiai/ssr_simple/model/TableCellModel.kt
package com.cincinnatiai.ssr_simple.model

data class TableCellModel(
    val text: String,
    val textStyle: TextStyleModel? = null, // Override column/table text style
    val backgroundColor: String? = null, // Cell background color
    val action: String? = null, // Click action for this cell
    val modifier: ModifierModel? = null, // Additional modifiers for the cell
    val showBorder: Boolean? = null // Override table border setting for this cell
)

File: ssr-simple/ssr-simple-react/src/models/TableCellModel.ts
import { ModifierModel } from './ModifierModel';
import { TextStyleModel } from './TextStyleModel';

export interface TableCellModel {
  text: string;
  textStyle?: TextStyleModel;
  backgroundColor?: string;
  action?: string;
  modifier?: ModifierModel;
  showBorder?: boolean;
}
